CMR/python/test/cmr/auth/test.py

```diff
@@ -0,0 +1,129 @@
+    # Tests
+
+    def test_age(self):
+        """ test the function used to judge if a file is too old to use """
+        # pylint: disable=W0212
+        self.assertEqual(True, t._days_old_from_time(1))
+        self.assertEqual(False, t._days_old_from_time(86400, 86400))
+        self.assertEqual(False, t._days_old_from_time(86400, 86400, 1.0))
+        self.assertEqual(False, t._days_old_from_time(86400, 86400*2, 1.0))
+        self.assertEqual(True, t._days_old_from_time(86400, 86400*2+1, 1.0))
+        self.assertEqual(True, t._days_old_from_time(86400, 86400*3, 1.0))
+        self.assertEqual(False, t._days_old_from_time(86400, 86400*3, 2.0))
+        self.assertEqual(True, t._days_old_from_time(86400, 86400*4, 2.0))
+
+    def test_password(self):
+        """Test the password pass through function"""
+        pass_func = t.password("test")
+        self.assertEqual(pass_func("foo", {}), "test")
+        self.assertEqual(pass_func("", {}), "test")
+        self.assertEqual(pass_func(None, {}), "test")
+
+    @patch('urllib.request.urlopen')
+    def test_token(self, urlopen_mock):
+        """Test a valid login using the clear text password lambda"""
+        options = {'cache.token': False, 'client.address':'127.0.0.1'}
+        pfunc = t.password("Text-Password")
+        self.assertEqual ("Text-Password", pfunc("Test.User", options))
+
+        urlopen_mock.return_value = valid_login_response()
+        actual = t.token("Test.User", t.password("Text-Password"), options)
+        self.assertEqual('fake-token', actual)
+
+    @patch('urllib.request.urlopen')
+    @patch('cmr.auth.token._read_file')
+    def test_password_file(self, read_mock, urlopen_mock):
+        """Test a valid login using the password file lambda"""
+        user = "Test.User"
+        expected_password = "File-Password"
+        options = {'cache.token': False, 'client.address':'127.0.0.1'}
+        read_mock.return_value = expected_password
+        self.assertEqual (expected_password, t.password_file(user, options))
+
+        urlopen_mock.return_value = valid_login_response()
+        actual = str(t.token(user, t.password_file, options))
+        self.assertEqual ("fake-token", actual)
+
+    @patch('urllib.request.urlopen')
+    @patch('cmr.auth.token._execute_command')
+    def test_password_manager(self, cmd_mock, urlopen_mock):
+        """test a valid login using the password manager"""
+        user = "Test.User"
+        expected_password = "Secure-Password"
+        options = {'cache.token': False, 'client.address':'127.0.0.1'}
+        cmd_mock.return_value = expected_password
+        self.assertEqual(cmd_mock.return_value, t.password_manager(user, options))
+
+        urlopen_mock.return_value = valid_login_response()
+        actual = str(t.token(user, t.password_manager, options))
+        self.assertEqual ('fake-token', actual)
+
+    @patch('urllib.request.urlopen')
+    def test_bad_login(self, urlopen_mock):
```

Comment:
is this exhaustive? do we want a separate failure for each type of login?

Reply:
either the password is wrong or it is not, it should not matter how you got that password

Reply:
Ok, so its not exercising that code, its hard to tell.  It looks like we have a cache.token option that is always false.  Are we never testing when its true?

Reply:
I did not want to write a file while testing and then have to clean it up. Now that I see how we can do tests on github (through actions) I'm reconsidering these. I will write a ticket and handle this in the next round (search).